### PR TITLE
[cli] added documentation for rcp and route

### PR DESCRIPTION
**Pull request by kylorene**
Updated RCP command documentation in CLI.

Merged RCP CLI commands in documentation.

Updated region and route CLI.

Updated region and route.

Updated route.

Updated route add.

Updated parameters for route add.

Added parameters for router.

Added formatting for parameters.

Updated route command function.

Updated rcp version parameter.

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -6216,15 +6247,37 @@ template <> otError Interpreter::Process<Cmd("region")>(Arg aArgs[])
...
+     * @cli releaserouterid (routerid)
+     * @code
+     * releaserouterid 16
+     * Done
+     * @endcode
+     * @cparam releaserouterid [@ca{routerid}]
+     * @par api_copy
+     * #otThreadReleaseRouterId
+     */
+{
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6260]: Looks like this bracket got moved and is causing all the failures.  It should be on line 6264 (between template and return).

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -6193,9 +6203,29 @@ template <> otError Interpreter::Process<Cmd("rcp")>(Arg aArgs[])
...
+    /**
+     * @cli region
+     * @code
+     * region
+     * US
+     * Done
+     * @endcode
+     * @par api_copy
+     * #otPlatRadioGetRegion
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6215]: Move this block left to align with the "template" on line 6216

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -6216,15 +6247,37 @@ template <> otError Interpreter::Process<Cmd("region")>(Arg aArgs[])
...
+    /**
+     * @cli releaserouterid (routerid)
+     * @code
+     * releaserouterid 16
+     * Done
+     * @endcode
+     * @cparam releaserouterid [@ca{routerid}]
+     * @par api_copy
+     * #otThreadReleaseRouterId
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6259]: Move this between 6261 and 6262 (between #if and template), and move left to align with "template"

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -6216,15 +6247,37 @@ template <> otError Interpreter::Process<Cmd("region")>(Arg aArgs[])
...
+    /**
+     * @cli rloc16
+     * @code
+     * rloc16
+     * 0xdead
+     * Done
+     * @endcode
+     * @par api_copy
+     * #otThreadGetRloc16
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6278]: Move left to align with "template"

File: src/cli/cli.cpp [hunk 5]
```diff
@@ -6405,19 +6555,62 @@ template <> otError Interpreter::Process<Cmd("router")>(Arg aArgs[])
...
+     * @endcode
+     * @code routerdowngradethreshold 23
+     * Done
+     * @endcode
+     * @cparam routerdowngradethreshold [@ca{threshold}]
+     * @par
+     * Gets or sets the ROUTER_DOWNGRADE_THRESHOLD value.
+     * @sa otThreadGetRouterDowngradeThreshold
+     * @sa otThreadSetRouterDowngradeThreshold
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6572]: Move this above the "template" (line 6556) and move left to align with that

File: src/cli/cli.cpp [hunk 6]
```diff
@@ -6405,19 +6555,62 @@ template <> otError Interpreter::Process<Cmd("router")>(Arg aArgs[])
...
+         * @code
+         * routereligible disable
+         * Done
+         * @endcode
+         * @cparam routereligible [@ca{enable|disable}]
+         * @par api_copy
+         * #otThreadIsRouterEligible
+         * @par
+         * Enables or disables the router role.
+         */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6610]: Move left to align with "else" on line 6611

File: src/cli/cli.cpp [hunk 7]
```diff
@@ -6430,19 +6623,70 @@ template <> otError Interpreter::Process<Cmd("routereligible")>(Arg aArgs[])
...
+     * @code
+     * routerselectionjitter 120
+     * Done
+     * @endcode
+     * @cparam routerselectionjitter [@ca{jitter}]
+     * @par
+     * Gets or sets the ROUTER_SELECTION_JITTER value.
+     * @sa otThreadGetRouterSelectionJitter
+     * @sa otThreadSetRouterSelectionJitter
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6642]: Move before the "template" (line 6624) and move left to align with it

File: src/cli/cli.cpp [hunk 8]
```diff
@@ -6430,19 +6623,70 @@ template <> otError Interpreter::Process<Cmd("routereligible")>(Arg aArgs[])
...
+     * @code
+     * routerupgradethreshold 16
+     * Done
+     * @endcode
+     * @cparam routerupgradethreshold [@ca{threshold}]
+     * @par
+     * Gets or sets the ROUTER_UPGRADE_THRESHOLD value.
+     * @sa otThreadGetRouterUpgradeThreshold
+     * @sa otThreadSetRouterUpgradeThreshold
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6664]: Move before the "template" (line 6646) and move left to align with it

File: src/cli/cli.cpp [hunk 9]
```diff
@@ -6430,19 +6623,70 @@ template <> otError Interpreter::Process<Cmd("routereligible")>(Arg aArgs[])
...
+     * @code
+     * childrouterlinks 16
+     * Done
+     * @endcode
+     * @cparam childrouterlinks [@ca{links}]
+     * @par
+     * Gets or sets the MLE_CHILD_ROUTER_LINKS value.
+     * @sa otThreadGetChildRouterLinks
+     * @sa otThreadSetChildRouterLinks
+     */
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6686]: Move before the "template" (line 6668) and move left to align with it

File: src/cli/cli.cpp [hunk 10]
```diff
@@ -6430,19 +6623,70 @@ template <> otError Interpreter::Process<Cmd("routereligible")>(Arg aArgs[])
...
     return ProcessGetSet(aArgs, otThreadGetChildRouterLinks, otThreadSetChildRouterLinks);
 }
-
+
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6689]: Remove this blank line... try not to add/remove blank lines that already exist in the file, whatever is already there should be left as is

**Review by Vyrastas — CHANGES_REQUESTED** [review]
The actual comments seem fine, just some formatting issues to address to keep it consistent with everything that's already there.  Make sure to run the import to DevSite again after you make these changes to verify it's still good on that side.

File: src/cli/cli.cpp (hunk 5 above)
Comment (Vyrastas) [reply src/cli/cli.cpp:6571]: Still not left-aligned to template

File: src/cli/cli.cpp (hunk 7 above)
Comment (Vyrastas) [reply src/cli/cli.cpp:6640]: Still not left-aligned to template

File: src/cli/cli.cpp (hunk 8 above)
Comment (Vyrastas) [reply src/cli/cli.cpp:6662]: Still not left-aligned to template

File: src/cli/cli.cpp (hunk 9 above)
Comment (Vyrastas) [reply src/cli/cli.cpp:6684]: Still not left-aligned to template

File: src/cli/cli.cpp [hunk 11]
```diff
@@ -6437,9 +6440,10 @@ template <> otError Interpreter::Process<Cmd("router")>(Arg aArgs[])
      * 8 24 50
      * Done
      * @endcode
+     * @par api_copy
+     * #otThreadGetRouterInfo
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6443]: These 2 lines are supposed to be

```
* @sa otThreadGetRouterInfo
```

File: src/cli/cli.cpp [hunk 12]
```diff
@@ -6519,7 +6523,6 @@ template <> otError Interpreter::Process<Cmd("router")>(Arg aArgs[])
      * Age: 7
      * Done
      * @endcode
-     * @cparam router [@ca{id}]
```
Comment (Vyrastas) [inline src/cli/cli.cpp]: Add this line back

File: src/cli/cli.cpp [hunk 13]
```diff
@@ -6386,7 +6390,6 @@ template <> otError Interpreter::Process<Cmd("route")>(Arg aArgs[])
      * route remove 2001:dead:beef:cafe::/64
      * Done
      * @endcode
-     * @cparam route remove [@ca{prefix}]
```
Comment (Vyrastas) [inline src/cli/cli.cpp]: Add this line back

**Review by Vyrastas — CHANGES_REQUESTED** [review]
The formatting fixes reverted some of the recent changes you did before submitting the PR.  I've commented on them... please fix and re-run the reference import so we can validate again.

File: src/cli/cli.cpp [hunk 14]
```diff
@@ -6188,7 +6188,9 @@ template <> otError Interpreter::Process<Cmd("rcp")>(Arg aArgs[])
      * Done
      * @endcode
      * @par api_copy
-     * #otPlatRadioGetVersionString
+     * #otGetVersionString
+     * @par
+     * Print RCP version string.
```
[the remark this answers is not in the document]
Comment (Vyrastas) [reply src/cli/cli.cpp:6193]: This one still needs doing

File: src/cli/cli.cpp [hunk 15]
```diff
@@ -6193,9 +6205,29 @@ template <> otError Interpreter::Process<Cmd("rcp")>(Arg aArgs[])
 exit:
     return error;
 }
-
+/**
+ * @cli region
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6209]: Can you move this block down before the current line 6234

```
if (aArgs[0].IsEmpty())
```

File: src/cli/cli.cpp [hunk 16]
```diff
@@ -6193,9 +6205,29 @@ template <> otError Interpreter::Process<Cmd("rcp")>(Arg aArgs[])
...
+ * region
+ * US
+ * Done
+ * @endcode
+ * @par api_copy
+ * #otPlatRadioGetRegion
+ */
 template <> otError Interpreter::Process<Cmd("region")>(Arg aArgs[])
 {
+    /**
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6220]: Can you move this block down before the current line 6239 (the `else` block)

File: src/cli/cli.cpp [hunk 17]
```diff
@@ -6217,6 +6217,24 @@ template <> otError Interpreter::Process<Cmd("rcp")>(Arg aArgs[])
...
+    uint16_t regionCode;
+/**
+ * @cli region
+ * @code
+ * region
+ * US
+ * Done
+ * @endcode
+ * @par api_copy
+ * #otPlatRadioGetRegion
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6231]: This is duplicated (it's already before the template line).  Please remove

File: src/cli/cli.cpp [hunk 18]
```diff
@@ -6402,22 +6551,65 @@ template <> otError Interpreter::Process<Cmd("router")>(Arg aArgs[])
...
+     * routereligible enable
+     * Done 
+     * @endcode
+     * @code
+     * routereligible disable
+     * Done
+     * @endcode
+     * @cparam routereligible [@ca{enable|disable}]
+     * @par api_copy
+     * #otThreadIsRouterEligible
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6606]: This should be 

#otThreadSetRouterEligible

File: src/cli/cli.cpp [hunk 19]
```diff
@@ -6402,22 +6551,65 @@ template <> otError Interpreter::Process<Cmd("router")>(Arg aArgs[])
...
-
+    /**
+     * @cli routereligible
+     * @code
+     * routereligible
+     * Enabled
+     * Done
+     * @endcode
+     * @par api_copy
+     * #otThreadIsRouterEligible
```
Comment (Vyrastas) [inline src/cli/cli.cpp:6586]: These two lines:

```
* @par api_copy
* #otThreadIsRouterEligible
```

Should be just an sa

```
* @sa otThreadIsRouterEligible
```

File: src/cli/cli.cpp (hunk 18 above)
Comment (Vyrastas) [inline src/cli/cli.cpp:6606]: These two lines

```
* @par api_copy
* #otThreadIsRouterEligible
```

Should be 

```
* @sa otThreadSetRouterEligible
```

Comment (Vyrastas) [inline src/cli/cli.cpp:6606]: These two lines

```
* @par api_copy
* #otThreadIsRouterEligible
```

Should be 

```
* @sa otThreadSetRouterEligible
```

**Comment by Vyrastas** [comment]
Just need to run make pretty and this should be good

**Review by jwhui — APPROVED** [review]

**Pull request merged**